test-plans/cmd/ping/main.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/muxer/mplex"
+	"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+	noise "github.com/libp2p/go-libp2p/p2p/security/noise"
+	libp2ptls "github.com/libp2p/go-libp2p/p2p/security/tls"
+	libp2pquic "github.com/libp2p/go-libp2p/p2p/transport/quic"
+	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
+	"github.com/libp2p/go-libp2p/p2p/transport/websocket"
+	libp2pwebtransport "github.com/libp2p/go-libp2p/p2p/transport/webtransport"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func main() {
+	var (
+		transport     = os.Getenv("transport")
+		secureChannel = os.Getenv("security")
+		muxer         = os.Getenv("muxer")
+		isDialerStr   = os.Getenv("is_dialer")
+		ip            = os.Getenv("ip")
+		redis_addr    = os.Getenv("REDIS_ADDR")
+	)
+
+	if redis_addr == "" {
+		redis_addr = "redis:6379"
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	// Get peer information via redis
+	rClient := redis.NewClient(&redis.Options{
+		DialTimeout: 10 * time.Second,
+		Addr:        redis_addr,
+		Password:    "",
+		DB:          0,
+	})
+	defer rClient.Close()
+
+	// Make sure redis is ready
+	_, err := rClient.Ping(ctx).Result()
+	if err != nil {
+		fmt.Println(err)
+		panic("Failed to connect to redis")
+	}
+
+	is_dialer := isDialerStr == "true"
+
+	options := []libp2p.Option{}
+
+	var listenAddr string
+	switch transport {
+	case "ws":
+		options = append(options, libp2p.Transport(websocket.New))
+		listenAddr = fmt.Sprintf("/ip4/%s/tcp/0/ws", ip)
+	case "tcp":
+		options = append(options, libp2p.Transport(tcp.NewTCPTransport))
+		listenAddr = fmt.Sprintf("/ip4/%s/tcp/0", ip)
+	case "quic":
+		options = append(options, libp2p.Transport(libp2pquic.NewTransport))
+		listenAddr = fmt.Sprintf("/ip4/%s/udp/0/quic", ip)
+	case "quic-v1":
+		options = append(options, libp2p.Transport(libp2pquic.NewTransport))
+		listenAddr = fmt.Sprintf("/ip4/%s/udp/0/quic-v1", ip)
+	case "webtransport":
+		options = append(options, libp2p.Transport(libp2pwebtransport.New))
+		listenAddr = fmt.Sprintf("/ip4/%s/udp/0/quic-v1/webtransport", ip)
+	default:
+		panic("Unsupported transport")
```

Comment:
```suggestion
		log.Fatalf("Unsupported transport: %s", transport)
```

Same for the `panic`s below.